server/services/offenderService.ts

```diff
@@ -71,8 +81,25 @@ export default function createOffenderService(elite2ClientBuilder): OffenderServ
   const getIncidentLocations = async (token: string, agencyId: AgencyId): Promise<PrisonLocation[]> => {
     try {
       const elite2Client = elite2ClientBuilder(token)
+      const incidentLocations = await elite2Client.getLocations(agencyId)
+
+      const primaryLocations = incidentLocations.filter(
+        loc =>
+          loc.userDescription.toUpperCase() === "PRISONER'S CELL" || loc.userDescription.toUpperCase() === 'OTHER CELL'
+      )
+      const remainingLocations = incidentLocations
+        .filter(
+          loc =>
+            loc.userDescription.toUpperCase() !== "PRISONER'S CELL" &&
+            loc.userDescription.toUpperCase() !== 'OTHER CELL'
+        )
+        .sort(compare)
+
+      if (primaryLocations.length === 2 && primaryLocations[0].userDescription.toUpperCase() !== "PRISONER'S CELL") {
+        primaryLocations.reverse()
+      }
```

Comment:
Having to handle this edge case isn't great.. would be good to avoid hardcoding a check of the first element and then having to mutate the array.  

An alternative would be to be more explicit with the finding and placing of the special elements:
```js
const  remainingLocations = ...

const prisonersCell = incidentLocations.find(...
const otherCell = incidentLocations.find(...

return [
...(prisonersCell ? [prisonersCell] : []), 
...(othersCell ? [othersCell] : []), 
...remainingIncidents,
]

Reply:
ordering by placing items in order in return array is much simpler! 
done.